server/gokbg3/grails-app/services/gokbg3/RestMappingService.groovy

```diff
@@ -320,7 +321,7 @@ class RestMappingService {
           }
         }
       } else {
-        def linkObj = ptype.get(val)
+        def linkObj = val.id?ptype.get(val.id):ptype.get(val)
```

Comment:
This will throw an exception for the integer case, since `val.id` is not defined.

Reply:
sorry. I wasn't aware of that change being pushed...